src/Files.App/Views/Properties/HashesPage.xaml

```diff
@@ -26,14 +26,68 @@
 
 	<Grid x:Name="RootGrid">
 
+		<!--  Hash Comparison Section  -->
 		<Grid
-			x:Name="HashesListGrid"
+			x:Name="HashComparisonGrid"
 			Margin="12"
 			VerticalAlignment="Top"
 			Background="{ThemeResource CardBackgroundFillColorDefaultBrush}"
 			BorderBrush="{ThemeResource CardStrokeColorDefaultBrush}"
 			BorderThickness="1"
 			CornerRadius="4">
+			<Grid.RowDefinitions>
+				<RowDefinition Height="Auto" />
+				<RowDefinition Height="Auto" />
+			</Grid.RowDefinitions>
+			<Grid.ColumnDefinitions>
+				<ColumnDefinition Width="*" />
+				<ColumnDefinition Width="Auto" />
+			</Grid.ColumnDefinitions>
+
+			<!--  Hash Input  -->
+			<Grid
+				Grid.Row="0"
+				Grid.Column="0"
+				Margin="12">
+				<Grid.ColumnDefinitions>
+					<ColumnDefinition Width="*" />
+				</Grid.ColumnDefinitions>
+
+				<TextBox
+					x:Name="HashInputTextBox"
+					Grid.Column="0"
+					PlaceholderText="{helpers:ResourceString Name=EnterHashToCompare}"
+					TextChanged="HashInputTextBox_TextChanged" />
+			</Grid>
+
+			<!--  Compare File Button  -->
+			<Button
+				x:Name="CompareFileButton"
+				Grid.Row="0"
+				Grid.Column="1"
+				Margin="12"
+				Click="CompareFileButton_Click"
+				Content="{helpers:ResourceString Name=CompareFile}" />
+		</Grid>
+
+		<InfoBar
+			x:Name="HashMatchInfoBar"
+			Grid.Row="1"
+			Grid.Column="0"
+			Margin="12,75,12,12"
+			IsClosable="False"
+			IsOpen="False"
```

Comment:
It might work better to set `IsOpen` to true, and bind `x:Load` to a property in the ViewModel.